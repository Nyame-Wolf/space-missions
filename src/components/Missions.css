.missions-page {
  margin-top: 90px;
  margin-bottom: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 90%;
  color: #000;
  font-family: var(--roboto);
}

table {
  width: 100%;
  border: 1px solid #aca9a970;
  margin-top: 20px;
  border-collapse: collapse;
  overflow: hidden;
}

tbody tr:nth-child(odd) {
  background-color: #d9d9d9;
  margin: 0;
}

button {
  font-family: var(--roboto);
  font-weight: 700;
  background-color: white;
  width: 90px;
  padding: 0.3rem;
  border: 1.5px solid #919191;
  border-radius: 0.15rem;
  margin: 1rem;
}

th {
  padding: 20px;
  font-size: 1.4rem;
  border: 1px solid #808080;
}

.name {
  font-weight: 800;
  font-size: 1.5rem;
  text-align: center;
  padding: 1rem;
}

.description {
  line-height: 1.2;
  font-size: 1rem;
  font-weight: 400;
  font-family: var(--monserrat);
  word-spacing: 2px;
  text-align: justify;
  padding: 1rem 1.2rem;
}

td {
  border-collapse: collapse;
  border: 1px solid #808080;
}

span {
  display: inline-block;
  width: 90px;
  border: 1.5px solid #000;
  margin: 0.8rem;
  color: white;
  font-size: 12px;
  font-weight: 700;
  text-align: center;
  border-radius: 0.15rem;
}

.not-member {
  background-color: #818181;
  padding: 8px;
}

.join {
  padding: 8px;
}

.join:hover {
  cursor: pointer;
  background-color: #b5f0d8;
}

.active-member {
  background-color: #21b4e5;
  padding: 8px;
}

.leave-mission {
  color: #c60101;
  border: 1px solid #4b0101;
  padding: 8px;
}

.leave-mission:hover {
  cursor: pointer;
  background-color: #efcccc;
}

@media (max-width: 600px) {
  .missions-page {
    display: unset;
  }

  th {
    font-size: 1rem;
    padding: 10px;
  }

  .name {
    font-size: 1rem;
  }

  .description {
    font-size: 1rem;
  }
}
