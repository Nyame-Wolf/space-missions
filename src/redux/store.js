import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import missionsSlice from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    // rockets: ,
    missions: missionsSlice,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
