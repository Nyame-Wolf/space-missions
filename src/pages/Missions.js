import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { addMissions, fetchMissions, removeMissions } from '../redux/missions/missionsSlice';
import MissionItem from '../components/MissionItem';
import '../components/Missions.css';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);

  useEffect(() => {
    if (!missions.length) {
      dispatch(fetchMissions());
    }
  }, []);

  const onAddMissions = (specificID) => {
    dispatch(addMissions(specificID));
  };

  const onRemoveMissions = (specificID) => {
    dispatch(removeMissions(specificID));
  };

  return (
    <div className="missions-page">
      <table>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>Join</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <MissionItem
              key={mission.id}
              id={mission.id}
              name={mission.name}
              description={mission.description}
              reserved={mission.reserved}
              onAddMission={onAddMissions}
              onRemoveMission={onRemoveMissions}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
