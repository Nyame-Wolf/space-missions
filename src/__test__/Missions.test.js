import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../pages/Missions';
import store from '../redux/store';

describe('Missions page component testing using snapshots', () => {
  it('Test Missions component renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
