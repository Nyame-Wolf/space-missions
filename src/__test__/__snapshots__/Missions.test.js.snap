// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions page component testing using snapshots Test Missions component renders 1`] = `
<div
  className="missions-page"
>
  <table>
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Join
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
